index.js

```diff
@@ -0,0 +1,74 @@
+import mysql from "mysql2/promise";
+import express from "express";
+
+async function init() {
+  const db = await mysql.createConnection({
+    host: "localhost",
+    user: "root",
+    database: "cook_assistant",
+    password: "admin",
+  });
+
+  const app = express();
+
+  app.set("view engine", "ejs");
+  app.use(express.urlencoded({ extended: true }));
+
+  app.get("/", async (_, res) => {
+    db.query("SELECT * FROM recipes").then(([recipes]) => {
+      res.render("index", { recipes });
+    });
+  });
+  app.post("/create-recipe", async (req, res) => {
+    const name = req.body.name;
+
+    db.query("INSERT INTO recipes (name) VALUES (?)", [name]).then(
+      ([{ insertId }]) => {
+        res.redirect("/recipe/" + insertId);
+      }
+    );
+  });
+  app.post("/delete-recipe/:id", async (req, res) => {
```

Comment:
I see you mix two different approaches to handle async function.
With using keyword async its better use await for db.query to handle request after query finished
or
not use async for function because you do not use await keyword. This will still work because you passing function to then() function that query runs after job is done.
Just my point of view.